### PR TITLE
Remove connection

### PR DESCRIPTION
**Pull request by emir4ik-code**
Created a generic DTO for connections CREATE, UPDATE and DELETE operations. 
Created new queryBuilder util deleteAndReturn

### CODE REVIEW

**Comment by emir4ik-code** [comment]
This PR depends on #135 Request/Accept connection. Approve after the latter if possible.

**Review by DLiSWE — DISMISSED** [review]

File: src/utils/queryBuilderUtils.ts [hunk 1]
```diff
@@ -6,3 +6,8 @@ export const updateAndReturn = async <ReturnType, PayloadType>(id: number, data:
   const updatedRow = await Entity.createQueryBuilder().update().where({ id }).set(data).returning('*').execute();
   return updatedRow.raw[0];
 };
+
+export const deleteAndReturn = async <ReturnType>(id: number, Entity: BaseEntityType): Promise<ReturnType> => {
+  const deletedRow = await Entity.createQueryBuilder().delete().where({ id }).returning('*').execute();
+  return deletedRow.raw[0];
```
Comment (Matt-Tan15) [inline src/utils/queryBuilderUtils.ts:12]: What does this return if you remove a connection?

File: src/controllers/connection.controller.ts [hunk 2]
```diff
@@ -3,13 +3,43 @@ import { RequestWithUser } from '@/interfaces';
...
+    } catch (error) {
+      next(error);
+    }
+  };
+  public sendConnectionRequest = async (req: RequestWithUser, res: Response, next: NextFunction) => {
+    try {
+      const senderId = req.user.id;
+      const receiverId = Number(req.params.user_id);
+      const data = await this.connectionService.sendConnectionRequest({ senderId, receiverId });
+      return res.status(201).json({ data });
```
Comment (Matt-Tan15) [inline src/controllers/connection.controller.ts:22]: We could send a message back saying a connection has been sent

File: src/controllers/connection.controller.ts [hunk 3]
```diff
@@ -3,13 +3,43 @@ import { RequestWithUser } from '@/interfaces';
...
+    } catch (error) {
+      next(error);
+    }
+  };
+  public removeConnection = async (req: RequestWithUser, res: Response, next: NextFunction) => {
+    try {
+      const senderId = req.user.id;
+      const receiverId = Number(req.params.user_id);
+      const data = await this.connectionService.removeConnection({ senderId, receiverId });
+      return res.status(200).json({ data });
```
Comment (Matt-Tan15) [inline src/controllers/connection.controller.ts:42]: let's add a message along with the data

**Review by Matt-Tan15 — APPROVED** [review]

**Review by DLiSWE — APPROVED** [review]

**Review by raylu511 — APPROVED** [review]

**Pull request merged**